ras_party/controllers/respondent_controller.py

```diff
@@ -131,11 +131,22 @@ def delete_respondents_marked_for_deletion(session):
     """
     respondents = session.query(Respondent).filter(Respondent.mark_for_deletion == True)  # noqa
     for respondent in respondents:
-        session.query(Enrolment).filter(Enrolment.respondent_id == respondent.id).delete()
-        session.query(BusinessRespondent).filter(BusinessRespondent.respondent_id == respondent.id).delete()
-        session.query(PendingEnrolment).filter(PendingEnrolment.respondent_id == respondent.id).delete()
-        session.query(Respondent).filter(Respondent.id == respondent.id).delete()
-        send_account_deletion_confirmation_email(respondent.email_address, respondent.first_name)
+        try:
+            session.query(Enrolment).filter(Enrolment.respondent_id == respondent.id).delete()
+            session.query(BusinessRespondent).filter(BusinessRespondent.respondent_id == respondent.id).delete()
+            session.query(PendingEnrolment).filter(PendingEnrolment.respondent_id == respondent.id).delete()
+            session.query(Respondent).filter(Respondent.id == respondent.id).delete()
+            send_account_deletion_confirmation_email(respondent.email_address, respondent.first_name)
```

Comment:
Nothing handled the any exceptions thrown in the above deletions. This makes trouble shooting errors impossible, and results in the user being told their account has been deleted. Technically it can't be used, but we need sight of any issues here. Might need to alert on these log entries.